### PR TITLE
[FE]: [Packs] Core [1/3]

### PR DESCRIPTION
**Pull request by alexmylonas**
## Background

This is the first PR for the `Packs` feature that includes some core changes and additions needed for the feature

## Changes

- Added `semver` package
- Updated schema
- Added graphql queries & fragments
- Backported `BulletLoading` component
- Added telemetry events

## Testing

N/A

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -140,6 +140,7 @@
     "react-json-view": "^1.19.1",
     "react-router": "^5.1.2",
     "react-router-dom": "^5.1.2",
+    "semver": "^7.3.4",
```
Comment (alexmylonas) [inline package.json:143]: I need this package to sort and compare arrays of versions

File: web/src/components/BulletedLoading/BulletedLoading.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ * All use, distribution, and/or modification of this software, whether commercial or non-commercial,
+ * falls under the Panther Commercial License to the extent it is permitted.
+ */
+
+import React from 'react';
+import { Box, BoxProps, useTheme } from 'pouncejs';
+
+type BulletedLoadingProps = BoxProps;
+
+const BulletedLoading: React.FC<BulletedLoadingProps> = props => {
```
Comment (alexmylonas) [inline web/src/components/BulletedLoading/BulletedLoading.tsx:15]: This is backported from enterprise

File: api/graphql/schema.graphql [hunk 3]
```diff
@@ -48,6 +48,7 @@ type Mutation {
   updateUser(input: UpdateUserInput!): User!
   uploadDetections(input: UploadDetectionsInput!): UploadDetectionsResponse
   updateGlobalPythonlModule(input: ModifyGlobalPythonModuleInput!): GlobalPythonModule!
+  updatePack(input: UpdatePackInput): Pack!
```
Comment (3nvi) [inline api/graphql/schema.graphql:51]:
```suggestion
  updatePack(input: UpdatePackInput!): Pack!
```

File: api/graphql/schema.graphql [hunk 4]
```diff
@@ -86,6 +88,30 @@ type Query {
   listCustomLogs: [CustomLogRecord!]!
 }
 
+input ListPacksInput {
+  ids: [String!]
```
Comment (3nvi) [inline api/graphql/schema.graphql:92]:
```suggestion
  ids: [ID!]
```

File: api/graphql/schema.graphql [hunk 5]
```diff
@@ -86,6 +88,30 @@ type Query {
...
+}
+
+input PackVersionInput {
+  id: String!
+  name: String!
+}
+
+input UpdatePackInput {
+  enabled: Boolean
+  id: String!
```
Comment (3nvi) [inline api/graphql/schema.graphql:111]:
```suggestion
  id: ID!
```

File: api/graphql/schema.graphql [hunk 6]
```diff
@@ -462,6 +492,43 @@ type Rule implements Detection {
...
 }
 
+type PackVersion {
+  id: String!
+  name: String!
+}
+
+type PackDetectionsPatterns {
+  # This is an array Detection['id'] that are related to the pack
+  IDs: [String!]
```
Comment (3nvi) [inline api/graphql/schema.graphql:502]:
```suggestion
  IDs: [ID!]
```

File: api/graphql/schema.graphql [hunk 7]
```diff
@@ -462,6 +492,43 @@ type Rule implements Detection {
...
+}
+
+type DetectionTypes {
+  GLOBAL: Int
+  RULE: Int
+  POLICY: Int
```
Comment (3nvi) [inline api/graphql/schema.graphql:508]: should those be required?

File: api/graphql/schema.graphql [hunk 8]
```diff
@@ -462,6 +492,43 @@ type Rule implements Detection {
...
+}
+
+type Pack {
+  id: String!
```
Comment (3nvi) [inline api/graphql/schema.graphql:512]:
```suggestion
  id: ID!
```

File: package.json (hunk 1 above)
Comment (3nvi) [reply package.json:143]: [9KB for just a comparison](https://bundlephobia.com/result?p=semver@7.3.4)? Seems like a lot... Can we write it on our own? If not, then omit this package from this PR and add itt again in the PR where you use it in order to see what can we do there

File: web/src/graphql/fragments/PackFull.graphql [hunk 9]
```diff
@@ -0,0 +1,30 @@
+fragment PackFull on Pack {
```
Comment (3nvi) [inline web/src/graphql/fragments/PackFull.graphql:1]: we should go into the habit of naming "complete" fragments with the suffix `...Details`. Can we name  this `PackDetails`?

**Review by 3nvi — COMMENTED** [review]
Looks nice @alexmylonas 🎉 . Let's do some updates & renames and decide on the `semver` package and we can move forwards

File: package.json (hunk 1 above)
Comment (alexmylonas) [reply package.json]: Removed!

**Review by 3nvi — APPROVED** [review]
Looks good @alexmylonas 🎉

**Pull request merged**